trie/trieStorageManager.go

```diff
@@ -371,8 +380,20 @@ func (tsm *trieStorageManager) TakeSnapshot(
 // SetCheckpoint creates a new checkpoint, or if there is another snapshot or checkpoint in progress,
 // it adds this checkpoint in the queue. The checkpoint operation creates a new snapshot file
 // only if there was no snapshot done prior to this
-func (tsm *trieStorageManager) SetCheckpoint(rootHash []byte, mainTrieRootHash []byte, leavesChan chan core.KeyValueHolder, stats common.SnapshotStatisticsHandler) {
-	if tsm.isClosed() {
+func (tsm *trieStorageManager) SetCheckpoint(
+	rootHash []byte,
+	mainTrieRootHash []byte,
+	leavesChan chan core.KeyValueHolder,
+	errChan chan error,
+	stats common.SnapshotStatisticsHandler,
+) {
+	if errChan == nil {
+		log.Error("programming error in trieStorageManager.SetCheckpoint, cannot set checkpoint because errChan is nil")
```

Comment:
write the error in the errChan? Valid for all error branches

Reply:
Do not write the err in errChan, because errChan is nil :sweat_smile:

Reply:
yes, in this case you are right. But the other cases?